src/network/networkd-ndisc.c

```diff
@@ -300,6 +305,8 @@ static int ndisc_router_process_route(Link *link, sd_ndisc_router *rt) {
 
         route->family = AF_INET6;
         route->table = link->network->ipv6_accept_ra_route_table;
+        if (!link->network->ipv6_accept_ra_route_table_set && link->network->vrf)
+                route->table = VRF(link->network->vrf)->table;
```

Comment:
maybe add a helper macro/inline function NETWORK_VRF_TABLE() or so that checks for NULL as necessary, and dereives the table from the Network object in one go?

Reply:
Not inline or macro, but I've introduced helper functions in networkd-link.c.